src/main/java/pingis/services/sandbox/SandboxService.java

```diff
@@ -46,7 +46,7 @@
   public boolean updateSubmissionResult(UUID submissionId, String testOutput, String stdout,
       String stderr, String validations, String vmLog, String status, int exitCode)
       throws IOException {
-    Submission submission = submissionRepository.findOne(submissionId);
+    Submission submission = submissionRepository.findById(submissionId).get();
 
     if (submission == null || submission.getStatus() != SubmissionStatus.PENDING) {
```

Comment:
Given that the repositories have moved to using `Optional<T>` I think it might be better to skip the `.get()` above and use `submission.isPresent()` instead of `submission == null` here.